tracer/src/Datadog.Trace/Baggage.cs

```diff
@@ -0,0 +1,357 @@
+﻿// <copyright file="Baggage.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+
+#nullable enable
+
+namespace Datadog.Trace;
+
+/// <summary>
+/// Baggage is a collection of name-value pairs that are propagated to downstream services.
+/// </summary>
+internal class Baggage : IDictionary<string, string>
```

Comment:
What's the motivation for having this extend `IDictionary<string,string>`?

Reply:
We need to expose baggage in `Datadog.Trace.Manual` and exposing it as a BCL type means we don't need duck-typing. `IDictionary<string, string>` has all the APIs we need to expose (get, set, remove, etc).